from PyQt5 import QtCore, QtWidgets


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(800, 600)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setMinimumSize(QtCore.QSize(800, 600))
        self.centralwidget.setMaximumSize(QtCore.QSize(800, 600))
        self.centralwidget.setObjectName("centralwidget")
        self.states_lineedit = QtWidgets.QLineEdit(self.centralwidget)
        self.states_lineedit.setGeometry(QtCore.QRect(100, 20, 231, 32))
        self.states_lineedit.setObjectName("states_lineedit")
        self.states_label = QtWidgets.QLabel(self.centralwidget)
        self.states_label.setGeometry(QtCore.QRect(7, 30, 81, 20))
        self.states_label.setObjectName("states_label")
        self.symbols_lineedit = QtWidgets.QLineEdit(self.centralwidget)
        self.symbols_lineedit.setGeometry(QtCore.QRect(100, 80, 231, 32))
        self.symbols_lineedit.setText("")
        self.symbols_lineedit.setObjectName("symbols_lineedit")
        self.symbols_label = QtWidgets.QLabel(self.centralwidget)
        self.symbols_label.setGeometry(QtCore.QRect(7, 90, 81, 20))
        self.symbols_label.setObjectName("symbols_label")
        self.start_state_lineedit = QtWidgets.QLineEdit(self.centralwidget)
        self.start_state_lineedit.setGeometry(QtCore.QRect(100, 140, 231, 32))
        self.start_state_lineedit.setText("")
        self.start_state_lineedit.setObjectName("start_state_lineedit")
        self.start_state_label = QtWidgets.QLabel(self.centralwidget)
        self.start_state_label.setGeometry(QtCore.QRect(7, 150, 81, 20))
        self.start_state_label.setObjectName("start_state_label")
        self.end_states_lineedit = QtWidgets.QLineEdit(self.centralwidget)
        self.end_states_lineedit.setGeometry(QtCore.QRect(100, 200, 231, 32))
        self.end_states_lineedit.setText("")
        self.end_states_lineedit.setObjectName("end_states_lineedit")
        self.end_states_label = QtWidgets.QLabel(self.centralwidget)
        self.end_states_label.setGeometry(QtCore.QRect(7, 210, 81, 20))
        self.end_states_label.setObjectName("end_states_label")
        self.transitions_label = QtWidgets.QLabel(self.centralwidget)
        self.transitions_label.setGeometry(QtCore.QRect(10, 260, 81, 20))
        self.transitions_label.setObjectName("transitions_label")
        self.transitions_textedit = QtWidgets.QPlainTextEdit(self.centralwidget)
        self.transitions_textedit.setGeometry(QtCore.QRect(100, 260, 231, 321))
        self.transitions_textedit.setObjectName("transitions_textedit")
        self.dfa_output_label = QtWidgets.QLabel(self.centralwidget)
        self.dfa_output_label.setGeometry(QtCore.QRect(370, 60, 411, 20))
        self.dfa_output_label.setAlignment(QtCore.Qt.AlignCenter)
        self.dfa_output_label.setObjectName("dfa_output_label")
        self.dfa_output_textedit = QtWidgets.QPlainTextEdit(self.centralwidget)
        self.dfa_output_textedit.setGeometry(QtCore.QRect(370, 80, 421, 371))
        self.dfa_output_textedit.setReadOnly(True)
        self.dfa_output_textedit.setObjectName("dfa_output_textedit")
        self.error_lineedit = QtWidgets.QLineEdit(self.centralwidget)
        self.error_lineedit.setGeometry(QtCore.QRect(470, 460, 321, 32))
        self.error_lineedit.setFocusPolicy(QtCore.Qt.NoFocus)
        self.error_lineedit.setReadOnly(True)
        self.error_lineedit.setObjectName("error_lineedit")
        self.error_label = QtWidgets.QLabel(self.centralwidget)
        self.error_label.setGeometry(QtCore.QRect(377, 470, 81, 20))
        self.error_label.setObjectName("error_label")
        self.reset_button = QtWidgets.QPushButton(self.centralwidget)
        self.reset_button.setGeometry(QtCore.QRect(410, 550, 101, 34))
        self.reset_button.setObjectName("reset_button")
        self.step_button = QtWidgets.QPushButton(self.centralwidget)
        self.step_button.setGeometry(QtCore.QRect(530, 550, 101, 34))
        self.step_button.setObjectName("step_button")
        self.process_button = QtWidgets.QPushButton(self.centralwidget)
        self.process_button.setGeometry(QtCore.QRect(650, 550, 101, 34))
        self.process_button.setObjectName("process_button")
        self.dfa_state_label = QtWidgets.QLabel(self.centralwidget)
        self.dfa_state_label.setGeometry(QtCore.QRect(377, 510, 81, 20))
        self.dfa_state_label.setObjectName("dfa_state_label")
        self.dfa_state_lineedit = QtWidgets.QLineEdit(self.centralwidget)
        self.dfa_state_lineedit.setGeometry(QtCore.QRect(470, 500, 321, 32))
        self.dfa_state_lineedit.setFocusPolicy(QtCore.Qt.NoFocus)
        self.dfa_state_lineedit.setReadOnly(True)
        self.dfa_state_lineedit.setObjectName("dfa_state_lineedit")
        self.line = QtWidgets.QFrame(self.centralwidget)
        self.line.setGeometry(QtCore.QRect(340, -10, 20, 621))
        self.line.setFrameShape(QtWidgets.QFrame.VLine)
        self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line.setObjectName("line")
        self.input_label = QtWidgets.QLabel(self.centralwidget)
        self.input_label.setGeometry(QtCore.QRect(370, 20, 31, 18))
        self.input_label.setObjectName("input_label")
        self.input_lineedit = QtWidgets.QLineEdit(self.centralwidget)
        self.input_lineedit.setGeometry(QtCore.QRect(410, 10, 381, 32))
        self.input_lineedit.setObjectName("input_lineedit")
        MainWindow.setCentralWidget(self.centralwidget)

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
        self.states_label.setText(_translate("MainWindow", "states"))
        self.symbols_label.setText(_translate("MainWindow", "symbols"))
        self.start_state_label.setText(_translate("MainWindow", "start state"))
        self.end_states_label.setText(_translate("MainWindow", "end states"))
        self.transitions_label.setText(_translate("MainWindow", "transitions"))
        self.dfa_output_label.setText(_translate("MainWindow", "dfa output"))
        self.error_label.setText(_translate("MainWindow", "error"))
        self.reset_button.setText(_translate("MainWindow", "reset"))
        self.step_button.setText(_translate("MainWindow", "step"))
        self.process_button.setText(_translate("MainWindow", "process"))
        self.dfa_state_label.setText(_translate("MainWindow", "dfa state"))
        self.input_label.setText(_translate("MainWindow", "input"))
